readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/kafka/PartitionsPauseStateHandler.java

```diff
@@ -44,18 +44,21 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
     }
 
     public void pause(Class pausingEvent) {
-        synchronized (consumer) {
-            if (pausedEvents.isEmpty() && !currentlyAssignedPartitions.isEmpty()) consumer.pause(currentlyAssignedPartitions);
-
+        if (pausedEvents.isEmpty() && !currentlyAssignedPartitions.isEmpty()) {
```

Comment:
hmm, you're right.
However, onPartitionAssigned andonPartitionRevoked are called inside consumer.poll which is currently doing a synchronized(consumer).
So it should be safe, but I will modify the code.

Reply:
I just jumped in. onPartitionsRevoked and onPartitionsAssigned are called by the same thread than does the consumer.poll (actually onPartitionsRevoked and onPartitionsAssigned are part of the consumer.poll process). 
Was that the concern ?

Reply:
hmm maybe I was not clear enough :)
Methods `onParitionRevoked` &  `onPartitionsAssigned` are not the concerned. If those methods are called by the consumer.poll process or whatever and hence under `synchronized (consumer)` does not make it more safe. It may just say that why are you synchronizing on consumer inside those methods then...

My concern is about `pause`/`resume` methods in which we are reading into `currentlyAssignedPartitions` instance without any read barrier because of @mchataigner change.

Reply:
Ok. What's misleading is that there should not be any synchronized anywhere anymore since pause/resume are also intended to be called from the same thread.